tests/data_io/test_csv.py

```diff
@@ -153,50 +153,82 @@ def test_export_csv(
         assert test_df.fillna("").isin(compare_to_df.fillna("")).all(axis=0).all()
 
 
+@mock_api()
+def test_dataset_export_csv_delim_error():
+    client = utils.client.create(**CONFIG["toolbox_test_instance"])
+    sm_dataset_id = CONFIG["datasets"]["minimal_schema_mapping_unified_dataset"]
+    dataset = client.datasets.by_resource_id(sm_dataset_id)
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        filepath = Path(tempdir) / "tests/data_io/test_taxonomy_invalid_delim.csv"
+
+        with pytest.raises(ValueError):
+            csv.from_dataset(dataset, filepath, csv_delimiter="|", flatten_delimiter="|")
+
+
+@pytest.mark.parametrize("overwrite", [False])
+@mock_api()
+def test_dataset_overwrite_file(overwrite: bool):
+    client = utils.client.create(**CONFIG["toolbox_test_instance"])
+    sm_dataset_id = CONFIG["datasets"]["minimal_schema_mapping_unified_dataset"]
+    dataset = client.datasets.by_resource_id(sm_dataset_id)
+
+    filepath = os.path.join(
+        get_toolbox_root_dir(), f"tests/data_io/test_dataset_overwrite_{overwrite}.csv"
+    )
+
+    f = open(filepath, "w")
+    f.write("Temporary file")
+    f.close()
+
+    with pytest.raises(FileExistsError):
+        csv.from_dataset(
+            dataset, filepath, csv_delimiter=",", flatten_delimiter="|", overwrite=overwrite
+        )
+
+    os.remove(filepath)
```

Comment:
Can this instead be done using `tempfile.TemporaryDirectory()` to write in a temporary dir that will be automatically cleaned up?